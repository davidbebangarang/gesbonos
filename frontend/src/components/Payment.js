import React from "react";
import StripeCheckout from "react-stripe-checkout";
import { toast } from "react-toastify";
import axios from "axios";
import "react-toastify/dist/ReactToastify.css";
import {withRouter} from 'react-router-dom';


toast.configure();

class Payment extends React.Component { 


 render() { 
      
       const {history,idInstancia,publicKey}=this.props
       let style;
       let userLog = JSON.parse(localStorage.getItem("user"))

       const product = {
          title: this.props.title,
          price: this.props.price,   
          bonoId: this.props.idBono,
          userData: {
            name: this.props.name,
            full_name:this.props.full_name,
            email:userLog? userLog.email: "",
            //country:this.props.country,
            address:this.props.adress,
            field_1:this.props.provincia,
            field_2:this.props.postCode         
            }
        }; 
        
        if(this.props.control){
          style = {  
            opacity: "1",
            disabled : false
          };  
        }else{
          style = {
            pointerEvents: "none", 
            opacity: "0.4",
            disabled : true
          };  
        }
              
       async function handleToken(token) {
         
          const url = process.env.REACT_APP_RUTA_CHECKOUT;
          product.userData.email = token.email;
          const response = await axios.post(`${url}${idInstancia}`,{token, product}); 
          const { status } = response.data;
          console.log("Response:", response.data);
          if (status === "success") {
            toast("Compra realizada con Ã©xito!!!", { type: "success" });
            history.push(`/${idInstancia}`);
          } else {
            toast("Error en la compra!!!", { type: "error" });
          }     
      };

      console.log(publicKey)
       return (         
        <div style={style} >   
          <StripeCheckout
            stripeKey="pk_test_4WAw3i1QHZJVrCXbwKROnmVA00ddoTVBHX"
            token={handleToken}                
            amount={product.precio* 100}
            name={product.title}
            currency="EUR"
            email= {product.userData.email}            
          >
            <button  className="btn btn-block btn-primary" >Comprar</button>
          </StripeCheckout>
         </div>       
      );
 } 
}
export default withRouter (Payment)