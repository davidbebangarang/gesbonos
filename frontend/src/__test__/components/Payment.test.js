import React from 'react';
import { mount, shallow } from 'enzyme';
import { create }from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Payment from '../../components/Payment';

describe('<ViewCards />', () => {
  
  test('Render del componente Payment', () => {
    const payment = shallow(
      <ProviderMock>
        <Payment />
      </ProviderMock>,
    );
    expect(payment.length).toEqual(1);
  });
});
describe('Payment Snapshot', () => {
  test('Comprobar la UI del componente Footer', () => {
    const payment = create(
      <ProviderMock>
        <Payment />
      </ProviderMock>,
    );
    expect(payment.toJSON()).toMatchSnapshot();
  });
});