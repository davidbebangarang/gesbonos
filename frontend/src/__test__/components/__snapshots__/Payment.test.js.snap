// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Snapshot Comprobar la UI del componente Footer 1`] = `
<div>
  <span
    onClick={[Function]}
  >
    <button
      className="btn btn-block btn-primary"
    >
      Comprar
    </button>
  </span>
</div>
`;
